### ISSUE TITLE
Provide a looping points API

### ISSUE THREAD

**Issue by linkmauve** [nickbrowne/ears#11]
I have a project where I get looping start and end points in samples, but I haven’t found any good way to do so for now.  #2 could allow that feature, but it would be very brittle.  A better solution would be to be able to provide the points directly in the API, so that the music thread would handle looping directly and per sample.

Suggestions welcome, I can work on that if you want.

**Comment by nickbrowne** [comment]
Agreed, though as you may have noticed in #2, the way the buffers are currently filled in the existing music code is a bit troublesome, I've had a few attempts at cleaning it up and trying to make it more robust but never arrived at anything all that satisfying.

I think https://github.com/nickbrowne/ears/compare/master...nickbrowne:fix-music-buffer-bug came the closest, but looking back, I don't think making the offset time based was a good idea. I think allowing users to seek via frames (and potentially also allow specifying custom start and end frames) will allow enough flexibility without complicating that code too much.

Before adding any extra features to the music streaming I'd like to fix the existing bugs though, I'll try to extract just the bug fixes from that branch tonight and get it merged.

**Comment by nickbrowne** [comment]
Hey just a quick update on this, I've been a bit busy at work the past 2 weeks but things have calmed down now, I should have time to do this over the next few nights, sorry for the delay.
